cty/walk_test.go

```diff
@@ -149,3 +157,77 @@ func TestTransform(t *testing.T) {
 		}
 	}
 }
+
+type errorTransformer struct{}
+
+func (errorTransformer) Enter(p Path, v Value) (Value, error) {
+	return v, nil
+}
+
+func (errorTransformer) Exit(p Path, v Value) (Value, error) {
+	ty := v.Type()
+	if ty.IsPrimitiveType() {
+		return v, nil
+	}
+	return v, p.NewError(fmt.Errorf("expected primitive type, was %#v", ty))
+}
+
+func TestTransformWithTransformer_error(t *testing.T) {
+	val := ObjectVal(map[string]Value{
+		"string": StringVal("hello"),
+		"number": NumberIntVal(10),
+		"bool":   True,
+		"list":   ListVal([]Value{True}),
+	})
+
+	gotVal, err := TransformWithTransformer(val, errorTransformer{})
+	if gotVal != DynamicVal {
+		t.Fatalf("expected DynamicVal, got %#v", gotVal)
+	}
+	if err == nil {
+		t.Fatal("expected error, got nil")
+	}
+
+	pathError, ok := err.(PathError)
+	if !ok {
+		t.Fatalf("expected PathError, got %#v", err)
+	}
+
+	if got, want := pathError.Path, GetAttrPath("list"); !got.Equals(want) {
+		t.Errorf("wrong path\n got: %#v\nwant: %#v", got, want)
+	}
+}
+
+func TestTransform(t *testing.T) {
+	val := ObjectVal(map[string]Value{
+		"list": ListVal([]Value{True}),
+		"set":  SetVal([]Value{True, False}),
+		"map":  MapVal(map[string]Value{"a": True, "b": False}),
+		"object": ObjectVal(map[string]Value{
+			"a": True,
+			"b": ListVal([]Value{False, False, False}),
+		}),
+	})
+	wantVal := ObjectVal(map[string]Value{
+		"list": ListVal([]Value{False}),
+		"set":  SetVal([]Value{True, False}),
```

Comment:
I'm getting that the set value here will be the same because the True/False flip and the Set is unordered ... but for paranoia should we have more than one member in the List (and have them be different) so that we can see that they do indeed flip? (the all False/all True case is covered here)

Reply:
Yep, good idea! Done.